java/src/jmri/util/LocoAddressComparator.java

```diff
@@ -19,9 +19,9 @@ public LocoAddressComparator() {
     public int compare(LocoAddress l1, LocoAddress l2) {
          if( l1.getProtocol() == l2.getProtocol() ){
              // protocol is the same, compare the number fields
-             return (l1.getNumber() - l2.getNumber());
+             return Integer.signum(l1.getNumber() - l2.getNumber());
```

Comment:
Just out of curiosity, why isn't this `return Integer.compare(l1.getNumber(), l2.getNumber);`

Reply:
Comes down to whether -1,0,+1 is required or not.  The 1.8 JavaDoc for Integer.compare(..) says

> the value 0 if x == y; a value less than 0 if x < y; and a value greater than 0 if x > y

and doesn't guarantee -1,0,+1.

Reply:
Right. Forgot that.

Reply:
Doing a double-take on that `Integer.compare(..)` JavaDoc, I don't see how it's promising anything different from just doing a subtraction....